jsoncons defines BOOST_HAS_FLOAT128

jsoncons's compiler_support header is defining BOOST_HAS_FLOAT128. This can lead to some interesting compiler errors when mixing jsoncons and boost, depending on header order:

```
 /usr/include/boost/atomic/detail/type_traits/is_iec559.hpp:37:19: error: 'float128_type' is not a member of 'boost'
  struct is_iec559< boost::float128_type >
```

This is because Boost and jsoncons disagree about when to define this (Boost has some extra checks that have to pass), so if you include some boost headers, then include jsoncons, then include more boost headers, the boost config headers can decide that there isn't a float128 type, while jsoncons decides there is, and then further boost headers think there should be a boost::float128_type when there isn't.

**What compiler, architecture, and operating system?**

- Compiler: GCC 5.4.0
- Architecture: x64
- Operating system: Ubuntu 16.04

**What jsoncons library version?**

 - [x] Latest release v0.156.0
 - [x] master

--- a/include/jsoncons/config/compiler_support.hpp
+++ b/include/jsoncons/config/compiler_support.hpp
@@ -263,7 +263,7 @@
 #if (defined(linux) || defined(__linux) || defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)) && !defined(_CRAYC)
 #if (__clang_major__ >= 4) && defined(__has_include)
 #if __has_include(<quadmath.h>)
-#  define BOOST_HAS_FLOAT128
+#  define JSONCONS_HAS_FLOAT128
 #endif
 #endif
 #endif
@@ -271,7 +271,7 @@
 
 #if defined(__GNUC__)
 #if defined(_GLIBCXX_USE_FLOAT128) 
-# define BOOST_HAS_FLOAT128
+# define JSONCONS_HAS_FLOAT128
 #endif
 #endif
 
@@ -279,7 +279,7 @@
 #if (defined(linux) || defined(__linux) || defined(__linux__) || defined(__GNU__) || defined(__GLIBC__)) && !defined(_CRAYC)
 #if (__clang_major__ >= 4) && defined(__has_include)
 #if __has_include(<quadmath.h>)
-#  define BOOST_HAS_FLOAT128
+#  define JSONCONS_HAS_FLOAT128
 #endif
 #endif
 #endif

